python/cuml/cuml/preprocessing/onehotencoder_mg.py

```diff
@@ -36,6 +34,10 @@ def __init__(self, *, client=None, **kwargs):
 
     def _check_input_fit(self, X, is_categories=False):
         """Helper function to check input of fit within the multi-gpu model"""
+        import dask.array
+
+        from cuml.dask.common.dask_arr_utils import to_dask_cudf
```

Comment:
Why do this? It would be nice to not have `import dask` stuff in modules that aren't in `cuml.dask` but doesn't this just move when the error occurs (for people who don't have dask installed)? And for people who have dask installed, does it make a difference to import it here instead of at the top level (presumably if you import `OneHotEncoderMG` you will give it something to fit?!)

Reply:
The multigpu classes are basically implementation details of their dask wrappers. As I understand it, no user should be using `OneHotEncoderMG` directly, they should be using `cuml.dask.preprocessing.OneHotEncoder`. Since the `*MG` classes aren't imported by default, we could have left these as top-level imports, but moving them to local better signaled to me what was dask-specific.